plugins/pulp_python/plugins/distributors/steps.py

```diff
@@ -33,13 +33,12 @@ def process_main(self):
         """
         Publish all the python files themselves by creating the symlinks to the storage paths.
         """
-        for name, packages in _get_packages(self.get_conduit().repo_id).items():
-            for package in packages:
-                relative_path = _get_package_path(name, package['filename'])
-                symlink_path = os.path.join(self.parent.web_working_dir, relative_path)
-                if not os.path.exists(os.path.dirname(symlink_path)):
-                    os.makedirs(os.path.dirname(symlink_path))
-                os.symlink(package['storage_path'], symlink_path)
+        repo_id = self.get_conduit().repo_id
+        for package in models.Package.objects.packages_in_repo(repo_id):
+            symlink_path = os.path.join(self.parent.web_working_dir, 'packages', package.src_path)
+            if not os.path.exists(os.path.dirname(symlink_path)):
```

Comment:
Actually you might want to handle it with a try/except and check the OSError's errno, since there is a race condition here between two processes publishing repos with a share directory structure.